src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowsOutWalker.cs

```diff
@@ -127,6 +127,11 @@ private Symbol GetNodeSymbol(BoundNode node)
                             return ((BoundRecursivePattern)node).Variable as LocalSymbol;
                         }
 
+                    case BoundKind.ListPattern:
+                        {
+                            return ((BoundListPattern)node).Variable as LocalSymbol;
+                        }
```

Comment:
Are the changes in iteration 34-37 (DataFlowOutWalker, DefiniteAssignemtn and VariablesDeclaredWalker) all part of fixing test `ListPattern_Negated_01`?

Reply:
I looked around for RecursivePattern to see what's missed. see comments below.